js/visualizer.js

```diff
@@ -355,17 +355,31 @@ function addDonutChart(target, datum, criteria=[]) {
 
   g.append("text")
     .attr("dy", ".35em")
-    .attr("x", function (d) {
-      const textAngle = (d.endAngle + d.startAngle)/2;
-      return textRadius * Math.sin(textAngle);
+    .attr("transform", function(d) {
+      //this is where I want to make a translation to the outside border
+      var c = labelArc.centroid(d);
+      const translate = "translate(" + c[0] +"," + c[1] + ")";
+      const roatateAngle = ((d.startAngle + d.endAngle)/2 - Math.PI/2) * (180/Math.PI);
+      const rotate = "rotate("  + roatateAngle +  ")"; 
+      //This is to make sure the text on the left side of the donut
+      //Is right side up. 
+      const additionalRotate = roatateAngle >= 90 ? "rotate(180)" : "" ;
```

Comment:
This is some creative spelling. Can we just change `roatateAngle` wherever it occurs?